### ISSUE TITLE
Rocky's Boots Precise Movement Problem Solved!

### ISSUE THREAD

**Issue by curby** [dosbox-staging/dosbox-staging#3652]
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### What Operating System are you using?

macOS (Apple Silicon)

### Whats your question and how can we help?

The DOS version of Rocky's Boots uses Shift and the arrow keys for precise movement commands. This doesn't work if you press the standard arrow keys. You can see it fail here (it also fails in dosbox-staging 0.81):

https://archive.org/details/msdos_Rockys_Boots_1985

The solution is to use the keymapper to have the arrow keys trigger numpad arrow key events in dosbox-staging.

[https://files.curby.net/rocky.map](Sample map file)

The key lines:
```
key_up
key_left
key_down
key_right
key_kp_8 "key 96" "key 82"
key_kp_4 "key 92" "key 80"
key_kp_6 "key 94" "key 79"
key_kp_2 "key 90" "key 81"
```

I'd like to add this to the wiki here. Should it go on the keymapper page or the configuration examples page? The keymapper page doesn't have game-specific mapping examples, but I think it'd help to include some, especially in cases like this where the keymapper is used to solve tricky control-input issues.

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

**Comment by johnnovak** [comment]
Thanks for the tip @curby!

> I'd like to add this to the wiki here. Should it go on the keymapper page or the configuration examples page?

Neither 😄 

It should be added to the [Game issues](https://github.com/dosbox-staging/dosbox-staging/wiki/Game-issues) page. I don't think you have access to editing the wiki pages, though. If you can't, let me know, and I'll add it.

**Comment by johnnovak** [comment]
But wait a minute... is this a Staging bug? Do Shift+Arrow keys work in other DOSBoxes or earlier versions of Staging? Can you perhaphs confirm the behaviour on real hardware too?

**Comment by curby** [comment]
I'm not sure if the original DOS release intended you to use the numpad's arrow keys, since the "normal" inverted-t arrows work for normal movement. It's also not seen only in -staging: I think archive.org uses vanilla Dosbox (linked above) and the problem exists there as well. 

Whether it's a bug is goes back to developer intent, so no idea. :P I'm just glad I found a work-around!

**Comment by interloper98** [comment]
XT PCs were common in the early to mid 80s, and most XT keyboards didn't have separate arrow keys: https://en.wikipedia.org/wiki/File:IBM_PC_XT_5160.JPG

IBM's AT PC was released just before 1985 when this game came out, but that PC cost $6000.  I think the bigger keyboards with separate arrow keys got more popular in the late 80s when people were buying AT PCs and clones.

**Comment by curby** [comment]
Yeah it's just odd that arrow keys still work for normal movement, just not precise movement.

**Comment by interloper98** [comment]
Indeed; can you track down the original manual? Would be interesting to see if this is mentioned.

**Comment by curby** [comment]
> It should be added to the [Game issues](https://github.com/dosbox-staging/dosbox-staging/wiki/Game-issues) page. 

Added! Feel free to rephrase/tweak as needed for consistency. :)

**Comment by johnnovak** [comment]
> > It should be added to the [Game issues](https://github.com/dosbox-staging/dosbox-staging/wiki/Game-issues) page.
> 
> Added! Feel free to rephrase/tweak as needed for consistency. :)

Nice one, thanks for adding it to the wiki @curby, the entry looks good 👍🏻 

Just one thing, could you please remove the link to that externally hosted file and convert it into a [GitHub Gist](https://gist.github.com/)? Thanks!

We'll leave the ticket open with the "bug" label, but it looks like we can't really test this on real hardware if we don't have access to a really old pre-1985 PC, which is highly unlikely 😄 My Pentium MMX won't help here 😄

**Comment by johnnovak** [comment]
DOSBox-X has the `controllertype = xt` setting which might help with this games and similar XT-era games.

https://github.com/joncampbell123/dosbox-x/blob/2d3dcc2325af3cb439cd8aa48308516dab8a47c3/src/dosbox.cpp#L3089-L3096

California Games has the same problem, and `controllertype = xt` seems to be just the ticket:
https://www.pcgamingwiki.com/wiki/California_Games

<img width="955" alt="image" src="https://github.com/dosbox-staging/dosbox-staging/assets/698770/05e78996-a082-4f90-8054-87d0ca6cb297">

Although, it seems the feature has regressed in X:
https://github.com/joncampbell123/dosbox-x/issues/4950

@interloper98 You might be perhaps interested in this?

**Comment by interloper98** [comment]
> Although, it seems the feature has regressed in X

Yes, looks like the conf setting was added in 2018, but I'm not seeing any implementation behind it since then -- maybe had to be deferred for more pressing issues :thinking:  :slightly_smiling_face:

**Comment by johnnovak** [comment]
> > Although, it seems the feature has regressed in X
> 
> Yes, looks like DOSBox-X's `controllertype = ` conf setting was added in 2018, but I'm not seeing any implementation behind it since then -- maybe had to be deferred for more pressing issues 🤔 🙂

Interesting. The PC Gaming Wiki article implies the feature was function at some point in DOSBox-X.

**Comment by interloper98** [comment]
> Interesting. The PC Gaming Wiki article implies the feature was functional at some point in DOSBox-X.

`git log -S controllertype` 

There are four hits:
 - Two are mass conf setting dumps (both in 2020)
 - Other two just add and modify this conf settings (both in 2018)

There's no code that reads or uses this setting.

Quite a mystery!

**Comment by johnnovak** [comment]
> There's no code that reads or uses this setting.
> 
> Quite a mystery!

Well, then they PC Gaming Wiki article is just plain wrong. Entirely within the realm of possibility 😏

False information on the internet—who has even heard of such a thing??! 😆

**Comment by interloper98** [comment]
I would believe it! The placebo effect works on gamers :-) 

"Researchers have discovered gamers have more fun when they think a video game has been updated with fancy new features"

https://www.newscientist.com/article/dn28319-placebo-effect-works-in-video-games-too/

Bro, you gotta try `"controllertype = xt-turbo-2000"`, it's next level.

**Comment by johnnovak** [comment]
> Bro, you gotta try `"controllertype = xt-turbo-2000"`, it's next level.

Maybe I should start advertising bogus settings in 0.82.0 🤔 😆 Well, I'm gonna add 96 kHz support in my current PR, so there we go, we're joining the placebo hype train 😎 

Or maybe just mention "AI-assisted" in every second sentence in the notes. That will do as well 🤣

**Comment by curby** [comment]
> Just one thing, could you please remove the link to that externally hosted file and convert it into a [GitHub Gist](https://gist.github.com/)? Thanks!

Done! 😊
